### PR TITLE
GG-31928 EVT_CLIENT_NODE_DISCONNECTED is not triggered in k8s

### PR DESCRIPTION
**Pull request by ashapkin**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoveryIpFinderFailureTest.java [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+     * and joinTimeout is not specified.
+     * Server node should be constantly trying to obtain IP addresses.
+     */
+    @Test
+    public void testServerNodeBrokenDynamicIpFinderFromTheStartNoJoinTimeout() throws Exception {
+        IgniteConfiguration cfgSrv = getConfigurationDynamicIpFinder("Server1", false, 0);
+
+        dynamicIpFinder.breakService();
+
+        final Boolean[] done = {false};
```
Comment (ibessonov) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoveryIpFinderFailureTest.java:207]: You should use AtomicBoolean here, boolean array is not safe.

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoveryIpFinderFailureTest.java [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+     * and joinTimeout is set to 0
+     * Server node should be constantly trying to obtain IP addresses.
+     */
+    @Test
+    public void testServerNodeBrokenStaticIpFinderZeroJoinTimeout() throws Exception {
+        IgniteConfiguration cfgSrv = getConfigurationDynamicIpFinder("Server1", false, 0);
+        dynamicIpFinder.setShared(false);
+        dynamicIpFinder.breakService();
+
+        final Boolean[] done = {false};
```
Comment (ibessonov) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoveryIpFinderFailureTest.java:251]: as well as here

File: modules/kubernetes/src/test/java/org/apache/ignite/discovery/KubernetesDiscoveryAbstractTest.java [hunk 3]
```diff
@@ -101,42 +78,59 @@ public void testClientConnectsToCluster() throws Exception {
...
+     * Mock HTTP server responce
+     *
+     * @param times How many times mock should return value.
+     * @param addrs Address list to return.
+     */
+    protected final void mockServerResponse(int times, String... addrs) {
         String ipAddrs = Arrays.stream(addrs)
-            .map(addr -> String.format("{\"ip\":\"%s\"}", addr))
-            .collect(Collectors.joining(","));
+                .map(addr -> String.format("{\"ip\":\"%s\"}", addr))
```
Comment (ibessonov) [inline modules/kubernetes/src/test/java/org/apache/ignite/discovery/KubernetesDiscoveryAbstractTest.java:106]: What's the reason behind reformatting? 4 spaces is a proper padding according to our code style guidelines.

File: modules/kubernetes/src/test/java/org/apache/ignite/discovery/KubernetesDiscoveryAbstractTest.java [hunk 4]
```diff
@@ -153,4 +147,4 @@ private KubernetesConnectionConfiguration prepareConfiguration()
 
         return cfg;
     }
-}
+}
```
Comment (ibessonov) [inline modules/kubernetes/src/test/java/org/apache/ignite/discovery/KubernetesDiscoveryAbstractTest.java:150]: Empty line required

Comment (ashapkin) [reply modules/kubernetes/src/test/java/org/apache/ignite/discovery/KubernetesDiscoveryAbstractTest.java]: Done

**Review by ibessonov — APPROVED** [review]

**Pull request merged**